api/samplepr.js

```diff
@@ -1,28 +1,119 @@
-const { spawnSync } = require('child_process');
+const util = require('util');
+const exec = util.promisify(require('child_process').exec);
+const datum = require('./testdatajson');
 const fs = require('fs');
+const path = require('path');
 
-function createPR(absolutePath, loc, username, currTime, title) {
-  const branchName = loc.replace('/', '-');
-  const initializeResult = spawnSync('bash', ['initializeRepo.sh', absolutePath, loc, branchName, username, currTime]);
-  if (initializeResult.status !== 0) {
-    console.error(`Error occurred while running initializeRepo.sh. Output: ${initializeResult.stderr.toString()}`);
-    return;
-  }
 
-  const githubResult = spawnSync('bash', ['github.sh', branchName, username, currTime, title]);
-  if (githubResult.status !== 0) {
-    console.error(`Error occurred while creating pull request. Output: ${githubResult.stderr.toString()}`);
-    return;
+const result = datum.body;
+const url = datum.media;
+const title = datum.title;
+const date = datum.date;
+const tags = datum.tags;
+const category = datum.categories;
+const speakers = datum.speakers;
+const username = "xyz";//"github-username"; <-- uncomment this and replace with your authenticated username to test. this will later be replace with the reviewer's github username
+const pr = true;
+
+
+function writeToFile(result, url, title, date, tags, category, speakers, videoTitle, username, local, test, pr) {
+  try {
+    let transcribedText = result;
+    let fileTitle;
+    if (title) {
+      fileTitle = title;
+    } else {
+      fileTitle = videoTitle;
+    }
+
+    let metaData = `---\n` +
+                   `title: ${fileTitle}\n` +
+                   `transcript_by: ${username} via TBTBTC v3\n`;
+
+    if (!local) {
+      metaData += `media: ${url}\n`;
+    }
+
+    if (tags) {
+      tags = tags.trim();
+      tags = tags.split(",");
+      for (let i = 0; i < tags.length; i++) {
+        tags[i] = tags[i].trim();
+      }
+      metaData += `tags: ${tags}\n`;
+    }
+
+    if (speakers) {
+      speakers = speakers.trim();
+      speakers = speakers.split(",");
```

Comment:
For data that are iterable e.g speakers, categories, tags. There isn't a consensus to how they are represented. On my local DB, some are array strings `speakers =  '["John", "Franck"]'` when formatted they are not arrays but strings. Some are plain strings with `,` as a delimiter e.g `"John, Frank"` e.t.c

I think we should simply use arrays for iterable data, converting it to string with a `,` would be easier and more structured than the other way around. 
Note: On the frontend I anticipate these cases and turn them to arrays

Reply:
> I think we should simply use arrays for iterable data

+1